### DISCUSSION TITLE
Solved my own issue with getting .NET (core/8.0) API that uses CefSharp running properly on IIS Server (was not initializing web browser)

### DISCUSSION THREAD

**Discussion by shiggins-pmc** [cefsharp/CefSharp#5007] [General]
PSA for anyone having issues hosting a CefSharp project from an IIS server, experiencing an issue in which the ChromiumWebBrowser never finishes initializing (but it does initialize in your development environment): You may need to set your IIS Application Pool to "Load User Profile".
 
You can do so by opening up "Internet Information Services (IIS) Manager" on the server/hosting computer. In the "Connections" pane on the left, expand the server and then click on "Application Pools". In the main window labelled "Application Pools", right-click on the application pool that is running your CefSharp application, then click Advanced Settings. In the Advanced Settings window, find the Load User Model setting under the Process Model section, and set it to TRUE.
 
More on our problem for context:
Our PDF-generating API (using CefSharp.Offscreen) was working just fine when running locally, but once published to an IIS Server on our Azure virtual machine, it would hang indefinitely while waiting for browser.IsBrowserInitialized to be set to true after instantiating the ChromiumWebBrowser class and creating a browser (automatic by default, or set automaticallyCreateBrowser: false and then use [browserObject].CreateBrowser()). We tried everything we could think of before we eventually tried modifying our server configuration. Hopefully this post helps somebody else out there, as we've been battling this issue for two weeks and were ecstatic to finally find a solution.

The one thing is: I wish that we know WHY this change worked. Does CefSharp require a certain level of permissions to function properly with CEF, or is it something else about the User Model that's needed to make a ChromiumWebBrowser instance? I'd love to understand this better. If anybody has thoughts on the subject, please feel free to comment.

In addition, does anybody know if this is a security risk to allow our application to load a user model? What are the risks involved in doing this?

**Comment by amaitland** [comment]
> Does CefSharp require a certain level of permissions to function properly with CEF

It's not actually headless `Chromium`, it's basically the desktop instance that renders to a buffer. `Headless Chromium` wasn't a thing when `CefSharp.OffScreen` was first released. `CEF` doesn't support `Headless Chromium`.

https://github.com/hardkoded/puppeteer-sharp might be a better choice for your scenario.

**Reply by shiggins-pmc** [reply]
Thank you for the suggestion. Perhaps we'll go that route next time we run into an issue with CefSharp that we can't easily resolve.
